src/prototype_rust_sdk/workflow_context.rs

```diff
@@ -218,21 +220,64 @@ impl WfContext {
     }
 }
 
-struct WFCommandFut {
+struct WFCommandFut<T> {
+    cancel_command_id: Option<CommandID>,
     result_rx: oneshot::Receiver<UnblockEvent>,
+    mapper: Box<dyn Fn(UnblockEvent) -> T + Send + Sync + 'static>,
```

Comment:
Why move the mapping into here at all? This requires a Box now where one was not needed previously, and doesn't really seem to save much.

There's an alternative way to do this though that I think might be worth doing, where `Future` is not implemented for  `WFCommandFut<T>` but rather is implemented for `WFCommandFut<T> where T: From<UnblockEvent>` and then all we need to do is just implement `From` for the different output types. That way the mapping stuff moves out of each command instantiation and is nicely contained, and there's no boxing.